### PR TITLE
Ordinal fee separation && refactoring

### PR DESCRIPTION
**Pull request by uns1nkableSam**

### CODE REVIEW

File: src/inscribeit/create_inscription.cpp [hunk 1]
```diff
@@ -331,13 +339,14 @@ void CreateInscriptionBuilder::RestoreTransactions()
     genesis_pubkeyscript << 1;
     genesis_pubkeyscript << m_destination_pk.value();
 
-    CMutableTransaction funding_tx;
+    CMutableTransaction funding_tx = CreateFundingTxTemplate();
     funding_tx.vin = {CTxIn(COutPoint(uint256S(*m_txid), *m_nout))};
     funding_tx.vin.front().scriptWitness.stack.push_back(m_utxo_sig.value());
+
     funding_tx.vout = {CTxOut(*m_amount, funding_pubkeyscript)};
```
Comment (l2xl) [inline src/inscribeit/create_inscription.cpp]: Do not replace the whole object and assign its fields instead

File: src/inscribeit/swap_inscription.cpp [hunk 2]
```diff
@@ -788,6 +823,14 @@ SwapInscriptionBuilder &SwapInscriptionBuilder::OrdUTXO(const string &txid, uint
     return *this;
 }
 
+CAmount SwapInscriptionBuilder::getWholeFee(CAmount fee_rate) {
```
Comment (l2xl) [inline src/inscribeit/swap_inscription.cpp]: OrdCommitTx has an individual fee_rate. SwapInscriptionBuilder::getWholeFee() should take it into an account.

File: src/inscribeit/contract_builder.hpp [hunk 3]
```diff
@@ -55,6 +55,9 @@ class ContractBuilder
     ContractBuilder& operator=(const ContractBuilder& ) = default;
     ContractBuilder& operator=(ContractBuilder&& ) noexcept = default;
 
+    virtual std::vector<CMutableTransaction> getTransactions() = 0;
+    virtual CAmount getWholeFee(CAmount fee_rate);
```
Comment (l2xl) [inline src/inscribeit/contract_builder.hpp]: I suggest removing the fee_rate from arguments and using the one already assigned to the builder

**Review by l2xl — CHANGES_REQUESTED** [review]
Need to fix mining fee estimation for Ord Commit transaction

File: src/inscribeit/create_inscription.cpp (hunk 1 above)
Comment (uns1nkableSam) [reply src/inscribeit/create_inscription.cpp]: Done

File: src/inscribeit/swap_inscription.cpp (hunk 2 above)
Comment (uns1nkableSam) [reply src/inscribeit/swap_inscription.cpp]: Done

File: src/inscribeit/contract_builder.hpp (hunk 3 above)
Comment (uns1nkableSam) [reply src/inscribeit/contract_builder.hpp]: done

File: src/inscribeit/contract_builder.hpp [hunk 4]
```diff
@@ -55,6 +55,9 @@ class ContractBuilder
     ContractBuilder& operator=(const ContractBuilder& ) = default;
     ContractBuilder& operator=(ContractBuilder&& ) noexcept = default;
 
+    virtual std::vector<std::pair<CAmount,CMutableTransaction>> getTransactions() = 0;
+    virtual CAmount getWholeFee();
```
Comment (l2xl) [inline src/inscribeit/contract_builder.hpp]: Lets follow existing naming convention: name getter methods with capital letter  Get...()

File: src/inscribeit/contract_builder.cpp [hunk 5]
```diff
@@ -7,4 +7,11 @@ const std::string ContractBuilder::name_params = "params";
 const std::string ContractBuilder::name_version = "protocol_version";
 const std::string ContractBuilder::name_mining_fee_rate = "mining_fee_rate";
 
+CAmount ContractBuilder::getWholeFee() {
```
Comment (l2xl) [inline src/inscribeit/contract_builder.cpp]: GetWholeFee() should return string

**Review by l2xl — COMMENTED** [review]
Lets add method
```
string ContractBuilder::GetMinFundingAmount()
```

**Comment by l2xl** [comment]
ORD_AMOUNT = 5000
string ContractBuilder::GetMinFundingAmount():
* CreateInscription: ORD_AMOUNT + GetWholeFee()
* SwapINscription: OrdPrice + MarketFee + GetWholeFee

**Comment by l2xl** [comment]
Add test when a change output is less then dust limit (5sat) and add check to SwapInscriptionBuilder and do not create the change.

**Pull request merged**